diff --git a/IlmBase/CMakeLists.txt b/IlmBase/CMakeLists.txt
index 3d2c7245..ab4733b9 100644
--- a/IlmBase/CMakeLists.txt
+++ b/IlmBase/CMakeLists.txt
@@ -10,6 +10,9 @@ SET(ILMBASE_VERSION_PATCH "1")
 SET(ILMBASE_VERSION ${ILMBASE_VERSION_MAJOR}.${ILMBASE_VERSION_MINOR}.${ILMBASE_VERSION_PATCH})
 SET(ILMBASE_VERSION_API ${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR})
 
+if(NOT MSVC)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wno-deprecated")
+endif()
 
 ENABLE_TESTING()
 
diff --git a/IlmBase/Half/CMakeLists.txt b/IlmBase/Half/CMakeLists.txt
index 6f9714dd..5f583aaa 100644
--- a/IlmBase/Half/CMakeLists.txt
+++ b/IlmBase/Half/CMakeLists.txt
@@ -3,8 +3,9 @@
 ADD_EXECUTABLE ( eLut eLut.cpp )
 
 ADD_CUSTOM_COMMAND (
-  TARGET eLut POST_BUILD
-  COMMAND eLut > ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
+  COMMAND eLut ARGS > ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
+  DEPENDS eLut
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 )
 SET_SOURCE_FILES_PROPERTIES(
@@ -14,8 +15,9 @@ SET_SOURCE_FILES_PROPERTIES(
 
 ADD_EXECUTABLE ( toFloat toFloat.cpp )
 ADD_CUSTOM_COMMAND (
-  TARGET toFloat POST_BUILD
-  COMMAND toFloat > ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+  COMMAND toFloat ARGS > ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+  DEPENDS toFloat
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
 )
 SET_SOURCE_FILES_PROPERTIES(
@@ -27,9 +29,7 @@ SET_SOURCE_FILES_PROPERTIES(
   half.cpp
   PROPERTIES
   OBJECT_DEPENDS
-  ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
-  OBJECT_DEPENDS
-  ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+   "${CMAKE_CURRENT_BINARY_DIR}/eLut.h;${CMAKE_CURRENT_BINARY_DIR}/toFloat.h"
   )
 
 IF(BUILD_SHARED_LIBS)
diff --git a/OpenEXR/CMakeLists.txt b/OpenEXR/CMakeLists.txt
index 8c0a3856..d39635de 100644
--- a/OpenEXR/CMakeLists.txt
+++ b/OpenEXR/CMakeLists.txt
@@ -9,6 +9,9 @@ SET(OPENEXR_VERSION_PATCH "0")
 SET(OPENEXR_VERSION ${OPENEXR_VERSION_MAJOR}.${OPENEXR_VERSION_MINOR}.${OPENEXR_VERSION_PATCH})
 SET(OPENEXR_VERSION_API ${OPENEXR_VERSION_MAJOR}_${OPENEXR_VERSION_MINOR})
 
+if(NOT MSVC)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wno-deprecated")
+endif()
 
 # enable the tests
 ENABLE_TESTING()
diff --git a/OpenEXR/IlmImf/ImfTiledMisc.cpp b/OpenEXR/IlmImf/ImfTiledMisc.cpp
index 0afe1490..bf3dfa4f 100644
--- a/OpenEXR/IlmImf/ImfTiledMisc.cpp
+++ b/OpenEXR/IlmImf/ImfTiledMisc.cpp
@@ -45,6 +45,7 @@
 #include <ImfChannelList.h>
 #include <ImfTileDescription.h>
 #include <algorithm>
+#include <limits>
 
 #include "ImfNamespace.h"
 
diff --git a/OpenEXR_Viewers/CMakeLists.txt b/OpenEXR_Viewers/CMakeLists.txt
index 5a21f540..81169443 100644
--- a/OpenEXR_Viewers/CMakeLists.txt
+++ b/OpenEXR_Viewers/CMakeLists.txt
@@ -9,6 +9,10 @@ SET(OPENEXR_VERSION_PATCH "0")
 SET(OPENEXR_VERSION ${OPENEXR_VERSION_MAJOR}.${OPENEXR_VERSION_MINOR}.${OPENEXR_VERSION_PATCH})
 SET(OPENEXR_VERSION_API ${OPENEXR_VERSION_MAJOR}_${OPENEXR_VERSION_MINOR})
 
+if(NOT MSVC)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wno-deprecated")
+endif()
+
 
 # distro building 
 SET(CPACK_PACKAGE_VERSION_MAJOR "${OPENEXR_VERSION_MAJOR}")
diff --git a/PyIlmBase/CMakeLists.txt b/PyIlmBase/CMakeLists.txt
index a76d1554..cadd36b9 100644
--- a/PyIlmBase/CMakeLists.txt
+++ b/PyIlmBase/CMakeLists.txt
@@ -9,6 +9,9 @@ SET(ILMBASE_VERSION_PATCH "0")
 SET(ILMBASE_VERSION ${ILMBASE_VERSION_MAJOR}.${ILMBASE_VERSION_MINOR}.${ILMBASE_VERSION_PATCH})
 SET(ILMBASE_VERSION_API ${ILMBASE_VERSION_MAJOR}_${ILMBASE_VERSION_MINOR})
 
+if(NOT MSVC)
+  SET(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++11 -Wno-deprecated")
+endif()
 
 # enable the tests
 ENABLE_TESTING()
