diff --git a/CMakeLists.txt b/CMakeLists.txt
index 39b80250..779e10d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -129,6 +129,11 @@ IF(HAVE_STD_CXX11)
     SET(CMAKE_CXX_FLAGS "-std=c++11 ${CMAKE_CXX_FLAGS}")
 ENDIF()
 
+if(NOT MSVC)
+    SET(CMAKE_C_FLAGS "-fcommon ${CMAKE_C_FLAGS}")
+    SET(CMAKE_CXX_FLAGS "-fcommon ${CMAKE_CXX_FLAGS}")
+endif()
+
 if(NOT WIN32)
     # Check if _POSIX_C_SOURCE and _XOPEN_SOURCE needs to be set for POSIX functions
     CHECK_SYMBOL_EXISTS(posix_memalign stdlib.h HAVE_POSIX_MEMALIGN_DEFAULT)

