diff -ur a/util/pl/VC-32.pl b/util/pl/VC-32.pl
--- a/util/pl/VC-32.pl	2022-02-05 19:00:42.121653068 +0100
+++ b/util/pl/VC-32.pl	2022-02-05 19:01:02.361702818 +0100
@@ -155,8 +155,8 @@
 	}
 
 # generate symbols.pdb unconditionally
-$app_cflag.=" /Zi /Fd\$(TMP_D)/app";
-$lib_cflag.=" /Zi /Fd\$(TMP_D)/lib";
+$app_cflag.=" /Fd\$(TMP_D)/app";
+$lib_cflag.=" /Fd\$(TMP_D)/lib";
 $lflags.=" /debug";
 
 $obj='.obj';
@@ -214,7 +214,7 @@
 		$asm='nasm -f win64 -DNEAR -Ox -g';
 		$afile='-o ';
 	} else {
-		$asm='ml64 /c /Cp /Cx /Zi';
+		$asm='ml64 /c /Cp /Cx';
 		$afile='/Fo';
 	}
 } elsif ($FLAVOR =~ /WIN64I/) {
@@ -228,7 +228,7 @@
 	$asmtype="win32n";
 	$afile='-o ';
 } else {
-	$asm='ml /nologo /Cp /coff /c /Cx /Zi';
+	$asm='ml /nologo /Cp /coff /c /Cx';
 	$afile='/Fo';
 	$asmtype="win32";
 }
